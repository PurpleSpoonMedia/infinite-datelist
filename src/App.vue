<template>
  <div id="app">
    <app-header></app-header>
    <div class="container">
      <router-view></router-view>
    </div>
  </div>
</template>

<script>
import Header from './components/Header';

export default {
  created() {
    this.$store.dispatch('fetchUsers');
  },
  components: {
    appHeader: Header,
  }
}
</script>
