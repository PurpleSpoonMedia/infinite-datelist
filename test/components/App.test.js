import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router'
import App from '../../src/App.vue';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueRouter);

describe('App.vue', () => {
  let actions, store;

  beforeEach(() => {
    actions = {
      fetchUsers: jest.fn(),
    }
    store = new Vuex.Store({
      state: {},
      actions
    })
  })

  test('should render App correctly', () => {
    const wrapper = shallowMount(App, {
      store,
      localVue
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
