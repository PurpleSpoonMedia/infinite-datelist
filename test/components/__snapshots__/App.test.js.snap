// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should render App correctly 1`] = `
<div id="app">
  <appheader-stub></appheader-stub>
  <div class="container">
    <router-view-stub name="default"></router-view-stub>
  </div>
</div>
`;
